ticdc: UT TestServerTLSWithCommonNameAndRotate is not stable

### What did you do?

run TestServerTLSWithCommonNameAndRotate

### What did you expect to see?

pass successfully

### What did you see instead?

Error: 3/13 08:03:47.889 +00:00] [ERROR] [server.go:298] ["http server error"] [error="[CDC:ErrServeHTTP]serve http error: mux: server closed"] [errorVerbose="[CDC:ErrServeHTTP]serve http error: mux: server closed\ngithub.com/pingcap/errors.AddStack\n\t/home/runner/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20231212100244-799fae176cfb/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\t/home/runner/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20231212100244-799fae176cfb/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/errors.WrapError\n\t/home/runner/work/tiflow/tiflow/pkg/errors/helper.go:34\ngithub.com/pingcap/tiflow/cdc/server.(*server).startStatusHTTP.func1\n\t/home/runner/work/tiflow/tiflow/cdc/server/server.go:298\nruntime.goexit\n\t/opt/hostedtoolcache/go/1.21.7/x64/src/runtime/asm_amd64.s:1650"] [stack="github.com/pingcap/tiflow/cdc/server.(*server).startStatusHTTP.func1\n\t/home/runner/work/tiflow/tiflow/cdc/server/server.go:298"]
    server_test.go:369: 
        	Error Trace:	/home/runner/work/tiflow/tiflow/cdc/server/server_test.go:369
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	            	github.com/pingcap/errors.AddStack
        	            		/home/runner/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20231212100244-799fae176cfb/errors.go:174
        	            	github.com/pingcap/errors.Trace
        	            		/home/runner/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20231212100244-799fae176cfb/juju_adaptor.go:15
        	            	github.com/pingcap/tiflow/pkg/retry.run
        	            		/home/runner/work/tiflow/tiflow/pkg/retry/retry_with_opt.go:48
        	            	github.com/pingcap/tiflow/pkg/retry.Do
        	            		/home/runner/work/tiflow/tiflow/pkg/retry/retry_with_opt.go:34
        	            	github.com/pingcap/tiflow/cdc/server.TestServerTLSWithCommonNameAndRotate.func4
        	            		/home/runner/work/tiflow/tiflow/cdc/server/server_test.go:328
        	            	github.com/pingcap/tiflow/cdc/server.TestServerTLSWithCommonNameAndRotate
        	            		/home/runner/work/tiflow/tiflow/cdc/server/server_test.go:369
        	            	testing.tRunner
        	            		/opt/hostedtoolcache/go/1.21.7/x64/src/testing/testing.go:1595
        	            	runtime.goexit
        	            		/opt/hostedtoolcache/go/1.21.7/x64/src/runtime/asm_amd64.s:1650
        	Test:       	TestServerTLSWithCommonNameAndRotate


### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
(paste TiCDC version here)
```

--- a/cdc/server/server_test.go
+++ b/cdc/server/server_test.go
@@ -285,7 +285,7 @@ func TestServerTLSWithCommonNameAndRotate(t *testing.T) {
 	}()
 
 	statusURL := fmt.Sprintf("https://%s/api/v1/status", addr)
-	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), 40*time.Second)
 	defer cancel()
 
 	var wg sync.WaitGroup

